### DISCUSSION TITLE
the brew still could not install 2.10.1

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [pdm-project/pdm#2403] [Q&A] [answered]
```
$ brew install pdm                                                                                                                                   ‹ruby-2.7.2›
==> Downloading https://ghcr.io/v2/homebrew/core/pdm/manifests/2.9.3
Already downloaded: /Users/dolphin/Library/Caches/Homebrew/downloads/b9b331e42261fca1679ab6232b33ae7dfa065daf50efcd7281b795875dec9031--pdm-2.9.3.bottle_manifest.json
==> Fetching pdm
==> Downloading https://ghcr.io/v2/homebrew/core/pdm/blobs/sha256:d7e320e2753a87ccb2a0253760926e2a9ba596d1d3dffd15be710345b0c5338b
Already downloaded: /Users/dolphin/Library/Caches/Homebrew/downloads/c143f1260bf311eb99511f1be49821d5916149199e54b633c5ea5a54b901e031--pdm--2.9.3.ventura.bottle.tar.gz
==> Pouring pdm--2.9.3.ventura.bottle.tar.gz
==> Caveats
zsh completions have been installed to:
  /usr/local/share/zsh/site-functions
==> Summary
🍺  /usr/local/Cellar/pdm/2.9.3: 949 files, 11.5MB
==> Running `brew cleanup pdm`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
(venv) (base) 

```

**Comment by pawamoy** [comment]
I suppose something needs to be updated on brew's side?

**Comment by frostming** [comment, marked as answer]
run brew update and try again. it's not pdm's fault

**Comment by baggiponte** [comment]
I would also like to add I _believe_ that installing pdm with brew could raise some issues. I don't know if brew creates separate virtual environments for every python dependency it adds, but it might be safer to `brew install pipx` and use `pipx` to install pdm and other python CLIs. `pipx` ensures they will be installed in a virtual environment just for them. Maybe brew does the same, though?

**Reply by frostming** [reply]
yes, brew also installs pdm in a virtualenv, just it packs and builds all dependencies itself, and some of the dependencies are brew formulae themselves and they must be included by `depends_on` key.

https://github.com/Homebrew/homebrew-core/blob/master/Formula/p/pdm.rb

**Reply by frostming** [reply]
The error happens when the PDM formula is migrated to `python@3.12` but not all of its dependencies upgrade at the same time. So there may be still dependencies on `python@3.11`. In this case they will pass the dependency check but can't be found in an `python@3.12` environment.

**Reply by baggiponte** [reply]
thanks for the explanation!
